packages/admin/src/table/filterbar/FilterBarActiveFilterBadge.tsx

```diff
@@ -4,21 +4,22 @@ import { useForm } from "react-final-form";
 
 export type CometAdminFilterBarActiveFilterBadgeClassKeys = "hasValueCount";
 
-export const dirtyFieldsCount = (values: Record<string, any>, registeredFields: string[] = []) => {
+export const dirtyFieldsCount = (values: Record<string, any>, registeredFields: string[] = [], fieldPath?: string) => {
     let count = 0;
-    Object.entries(values).forEach(([fieldName, fieldValue]) => {
+    Object.entries(values).forEach(([fieldKey, fieldValue]) => {
         if (!fieldValue) {
             return;
         }
 
+        const fieldName = fieldPath ? `${fieldPath}.${fieldKey}` : fieldKey;
         const isRegisteredField = registeredFields.includes(fieldName);
         if (Array.isArray(fieldValue) && isRegisteredField) {
             count += fieldValue.length;
         } else if (typeof fieldValue === "object") {
             if (Object.keys(fieldValue).length === 2 && "min" in fieldValue && "max" in fieldValue && isRegisteredField) {
                 count++;
             } else {
-                count += dirtyFieldsCount(fieldValue, registeredFields);
+                count += dirtyFieldsCount(fieldValue, registeredFields, fieldName);
```

Comment:
that can be done without extra parameter: for the recursion call, the values object gets stripped down to include one level less (`website: { url: "xxx", name: "yyy" }` gets `{ url: "xxx", name: "yyy" }` - so website is stripped.

you can also do the same for registeredFields (in the first call its `["website.url", "website.name"]` and for the recursion you remove one level and get `["url", "name"]` (obviously non website.* parts neet to be removed completely, as for values)

Reply:
but how can i tell which values are the correct ones then ? if i stripp it i loose the path information which means i have the problem with the uniqueness of the values again, ain't i ?

Reply:
Even if values is eg `{ website: { url: "xxx", name: "yyy" }, foo: { name: "blah" } }`, you pass into the recursion for website only website sub-values `{ url: "xxx", name: "yyy" }`, so `["url", "name"]` are unique.

Reply:
i pushed my new solution for the count method :)